sdk/iothub/src/service/requests/delete_identity_builder.rs

```diff
@@ -0,0 +1,54 @@
+use http::{Method, StatusCode};
+
+use crate::service::{IoTHubError, ServiceClient, API_VERSION};
+
+/// The DeleteIdentityBuilder is used to construct a request to delete a module or device identity.
+pub struct DeleteIdentityBuilder<'a> {
+    service_client: &'a ServiceClient,
+    if_match: String,
+    device_id: String,
+    module_id: Option<String>,
+}
+
+impl<'a> DeleteIdentityBuilder<'a> {
+    pub(crate) fn new(
+        service_client: &'a ServiceClient,
+        if_match: String,
```

Comment:
Is this really required?

Reply:
If this is about the `if_match`:
Based on the docs I would think not. (https://docs.microsoft.com/en-us/rest/api/iothub/service/modules/deleteidentity)
However, invoking this without including the `if_match` value in the header leads to a `PreconditionFailed` error. In theory we could hardcode the value of `if_match` to be: `*`, but not sure if that's desirable.